OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/SSAIPlayerViewController.m

```diff
@@ -0,0 +1,144 @@
+  [self addChildViewController:self.skinController];
+  [self.skinController.view setFrame:self.videoView.bounds];
+
+  // Load the video
+  [self.ooyalaPlayer setEmbedCode:self.embedCode];
+  
+  self.ssaiPlugin = [[OOSsaiPlugin alloc] init];
+  [self.ssaiPlugin registerPlayer:self.ooyalaPlayer];
+  
+  // In QA Mode , making textView visible
+  self.qaView.hidden = !self.qaModeEnabled;
+  NSString *path = [[NSBundle mainBundle] pathForResource:self.playerSelectionOption.adSetProvider
+                                                   ofType:@"json"];
+  NSString *content = [NSString stringWithContentsOfFile:path
+                                                encoding:NSUTF8StringEncoding
+                                                   error:NULL];
+  self.playerParams.text = content;
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(notificationHandler:)
+                                               name:nil
+                                             object:self.ooyalaPlayer];
+  
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(notificationHandler:)
+                                               name:nil
+                                             object:self.ssaiPlugin];
+  
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(notificationHandler:)
+                                               name:nil
+                                             object:self.skinController];
+}
+
+#pragma mark - Private functions
+
+- (void)notificationHandler:(NSNotification*)notification {
+  
+  // Ignore TimeChangedNotificiations for shorter logs
+  if ([notification.name isEqualToString:OOOoyalaPlayerTimeChangedNotification]) {
+    return;
+  }
+  
+  NSString *message = [NSString stringWithFormat:@"Notification Received: %@. state: %@. playhead: %f count: %d",
+                       [notification name],
+                       [OOOoyalaPlayer playerStateToString:[self.skinController.player state]],
+                       [self.skinController.player playheadTime], appDel.count];
+  NSLog(@"%@",message);
+  
+  // In QA Mode , adding notifications to the TextView
+  if (self.qaModeEnabled == YES) {
+    NSString *string = self.qaLogTextView.text;
+    NSString *appendString = [NSString stringWithFormat:@"%@ :::::::::: %@", string, message];
+    [self.qaLogTextView setText:appendString];
+  }
+  
+  appDel.count++;
+}
+
+#pragma mark - Actions
+
+- (IBAction)setParams:(UIButton *)sender {
+  NSString *params = self.playerParams.text;
+  if  ( [self.ssaiPlugin setParams:params] && [self.ssaiPlugin.player state] != OOOoyalaPlayerStatePlaying ) {
+    [self.ooyalaPlayer setEmbedCode:self.playerSelectionOption.embedCode];
+  }
+}
+
+-(void)dealloc{
+  [self.ssaiPlugin deregisterPlayer:self.ooyalaPlayer];
```

Comment:
I think it's worth or better registering SSAI when the view will appear or disappear. What do you think?

Reply:
done,